File: bindings/dotnet/src/Ironpress/IronpressException.cs
namespace Ironpress;

/// <summary>A categorized failure returned by the Ironpress renderer.</summary>
public sealed class IronpressException : Exception
{
    internal IronpressException(IronpressErrorKind kind, int nativeStatus, string message)
        : base(message)
    {
        Kind = kind;
        NativeStatus = nativeStatus;
    }

    /// <summary>Gets the stable failure category.</summary>
    public IronpressErrorKind Kind { get; }

    /// <summary>Gets the numeric status returned by the native ABI.</summary>
    public int NativeStatus { get; }
}

File: bindings/java/src/main/java/io/github/gastongouron/ironpress/IronpressException.java
package io.github.gastongouron.ironpress;

/** A categorized failure returned by the Ironpress renderer. */
public final class IronpressException extends RuntimeException {
  private static final long serialVersionUID = 1L;

  private final IronpressErrorKind kind;
  private final int nativeStatus;

  IronpressException(IronpressErrorKind kind, int nativeStatus, String message) {
    super(message);
    this.kind = kind;
    this.nativeStatus = nativeStatus;
  }

  /** Return the stable failure category. */
  public IronpressErrorKind getKind() {
    return kind;
  }

  /** Return the numeric status produced by the native ABI. */
  public int getNativeStatus() {
    return nativeStatus;
  }
}
